pkg/deploy/api/validation/validation_test.go

```diff
@@ -79,12 +79,43 @@ func TestValidateDeploymentConfigOK(t *testing.T) {
 	}
 }
 
+func TestValidateDeploymentConfigICTMissingImage(t *testing.T) {
+	errs := ValidateDeploymentConfig(&api.DeploymentConfig{
+		ObjectMeta: kapi.ObjectMeta{Name: "foo", Namespace: "bar"},
+		Spec: api.DeploymentConfigSpec{
+			Replicas: 1,
+			Triggers: []api.DeploymentTriggerPolicy{test.OkImageChangeTrigger()},
+			Selector: test.OkSelector(),
+			Strategy: test.OkStrategy(),
+			Template: test.OkPodTemplateMissingImage("container1"),
+		},
+	})
+
+	if len(errs) > 0 {
+		t.Errorf("Unexpected non-empty error list: %+v", errs)
+	}
+}
+
 func TestValidateDeploymentConfigMissingFields(t *testing.T) {
 	errorCases := map[string]struct {
 		DeploymentConfig api.DeploymentConfig
 		ErrorType        field.ErrorType
 		Field            string
 	}{
+		"empty container field": {
+			api.DeploymentConfig{
+				ObjectMeta: kapi.ObjectMeta{Name: "foo", Namespace: "bar"},
+				Spec: api.DeploymentConfigSpec{
+					Replicas: 1,
+					Triggers: []api.DeploymentTriggerPolicy{test.OkConfigChangeTrigger()},
+					Selector: test.OkSelector(),
+					Strategy: test.OkStrategy(),
+					Template: test.OkPodTemplateMissingImage("container1"),
+				},
+			},
```

Comment:
The difference is this one has a ConfigChangeTrigger, where as  the passing one above has an ImageChangeTrigger.